### PR TITLE
order: Add `Side` type MarshalJSON method

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Adds MarshalJSON method

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/order/order_test.go [hunk 1]
```diff
@@ -2123,6 +2123,16 @@ func TestSideUnmarshal(t *testing.T) {
...
 }
 
+func TestSideMarshal(t *testing.T) {
+	t.Parallel()
+	b, err := Buy.MarshalJSON()
+	assert.NoError(t, err, "Marshalling Side okay")
+	assert.Equal(t, `"BUY"`, string(b), "Correctly marshalled Side")
+	b, err = UnknownSide.MarshalJSON()
+	assert.NoError(t, err, "Marshalling Side okay")
+	assert.Equal(t, `"UNKNOWN"`, string(b), "Correctly marshalled Side")
```
Comment (gloriousCode) [inline exchanges/order/order_test.go:2133]: The text param for assertify is for when it fails, with the intent to highlight what is expected and is rendered to the output


I note that it's similar to the test above, but that can optionally be fixed too

File: exchanges/order/orders.go [hunk 2]
```diff
@@ -1115,6 +1115,10 @@ func (s *Side) UnmarshalJSON(data []byte) (err error) {
 	return
 }
 
+func (s Side) MarshalJSON() ([]byte, error) {
```
Comment (gloriousCode) [inline exchanges/order/orders.go:1118]: Fix yo lint

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Weekend ~Warrior~ Wario

File: exchanges/order/order_test.go (hunk 1 above)
Comment (shazbert) [reply exchanges/order/order_test.go]: I rm'd them it was from auto complete

**Review by gloriousCode — APPROVED** [review]
[tACK!](https://www.mariouniverse.com/wp-content/audio/mp/mp/wario/4.mp3)

File: exchanges/order/order_test.go [hunk 3]
```diff
@@ -2123,6 +2123,16 @@ func TestSideUnmarshal(t *testing.T) {
 	assert.ErrorAs(t, s.UnmarshalJSON([]byte(`14`)), &jErr, "non-string valid json is rejected")
 }
 
+func TestSideMarshal(t *testing.T) {
```
Comment (thrasher-) [inline exchanges/order/order_test.go:2126]:
```suggestion
func TestSideMarshalJSON(t *testing.T) {
```

**Review by thrasher- — CHANGES_REQUESTED** [review]
Then this looks GTG

**Review by thrasher- — APPROVED** [review]

**Pull request merged**